### PR TITLE
feat: link component storefront styles

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Refactored the component and introduced standard link styles.

closes [HRZ-2462](https://spryker.atlassian.net/browse/HRZ-2462)

[HRZ-2462]: https://spryker.atlassian.net/browse/HRZ-2462?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/content/link/link.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      .color=${color}
+      ?singleLine=${singleLine}
+      .icon=${icon}
+    >
+      ${this.renderLink()}
+    </oryx-link>`;
+  }
+
+  protected renderLink(): TemplateResult {
+    const { label, target } = this.$options();
```
Comment (Sinucid) [inline libs/domain/content/link/link.component.ts]:
```suggestion
    const { label, target, icon, button } = this.$options();
```

File: libs/domain/content/link/link.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    const icon = this.$options().button && this.$options().icon;
```
Comment (Sinucid) [inline libs/domain/content/link/link.component.ts]:
```suggestion
    const renderIcon = button && icon;
```

File: libs/domain/content/link/link.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    return html`
+      <a
+        href=${this.$link()}
+        aria-label=${ifDefined(label)}
+        target=${ifDefined(target)}
+        rel=${ifDefined(this.getRel())}
+        ><slot>
+          ${when(
+            icon,
+            () => html`<oryx-icon .type=${this.$options().icon}></oryx-icon>`
```
Comment (Sinucid) [inline libs/domain/content/link/link.component.ts]:
```suggestion
            renderIcon,
            () => html`<oryx-icon .type=${icon}></oryx-icon>`
```

File: libs/domain/content/link/link.component.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  protected renderLink(): TemplateResult {
+    const { label, target } = this.$options();
+
+    const icon = this.$options().button && this.$options().icon;
+
+    return html`
+      <a
+        href=${this.$link()}
```
Comment (Sinucid) [inline libs/domain/content/link/link.component.ts:46]: Should we not render the anchor if link === null?

File: libs/domain/content/link/link.model.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface ContentLinkContent {
+  text?: string;
+}
+
+export interface ContentLinkOptions {
+  url?: string;
+  type?: SemanticLinkType;
+  id?: string;
+  params?: Record<string, string>;
```
Comment (Sinucid) [inline libs/domain/content/link/link.model.ts:12]: Docs?

File: libs/platform/experience/src/color/colors/gray/gray.ts [hunk 6]
```diff
@@ -2,7 +2,8 @@ import { Color } from '../../color.model';
 
 export const gray: Color = {
   light: {
-    0: 'hsl(0, 0%, 100.0%)',
+    // 0: 'hsl(0, 0%, 100.0%)',
```
Comment (Sinucid) [inline libs/platform/experience/src/color/colors/gray/gray.ts]: leftover?

File: libs/platform/experience/src/color/colors/gray/gray.ts [hunk 7]
```diff
@@ -17,7 +18,8 @@ export const gray: Color = {
     12: 'hsl(0, 0%, 9.0%)',
   },
   dark: {
-    0: 'hsl(0, 0%, 0%)',
+    // 0: 'hsl(0, 0%, 0%)',
```
Comment (Sinucid) [inline libs/platform/experience/src/color/colors/gray/gray.ts]: the same

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/content/link/link.model.ts (hunk 5 above)
Comment (tobi-or-not-tobi) [reply libs/domain/content/link/link.model.ts:12]: we likely do another refactor on those, so I left docs out for now

File: libs/base/ui/action/link/src/link.component.ts [hunk 8]
```diff
@@ -1,29 +1,33 @@
...
   @property({ type: Boolean, reflect: true }) disabled?: boolean;
-  @property({ reflect: true }) icon?: Icons | string;
-  @property({ reflect: true }) multiLine?: boolean;
+  @property({ reflect: true }) singleLine?: boolean;
+
+  protected iconType?: Icons | string;
+
+  @property() set icon(value: Icons | string) {
+    this.iconType = value;
+    this.toggleAttribute('has-icon', !!value);
```
Comment (Sinucid) [inline libs/base/ui/action/link/src/link.component.ts]: Why do you need this setter? Just to set the 'has-icon' attr? Try the selector like `[icon]:not([icon=""])`

**Review by Sinucid — CHANGES_REQUESTED** [review]

**Review by Sinucid — DISMISSED** [review]

**Review by Sinucid — APPROVED** [review]

**Pull request merged**